controllers/istio/istio_controller.go

```diff
@@ -105,8 +106,12 @@ func validate(istio *v1.Istio) error {
 }
 
 func (r *Reconciler) reconcileActiveRevision(ctx context.Context, istio *v1.Istio) error {
+	version, err := istioversions.ResolveVersion(istio.Spec.Version)
+	if err != nil {
+		return fmt.Errorf("version %q not found: %w", istio.Spec.Version, err)
```

Comment:
The error returned by `ResolveVersion` already contains the same text ("version xyz not found"). 

In general, I think the original function should add the information that it can; the calling function should just add any additional context that the called function can't know. 

Here, the `reconcileActiveRevision` function should report the `Istio.metadata.name` in the error.